### DISCUSSION TITLE
Representative sequences for tax4fun 2 analysis

### DISCUSSION THREAD

**Discussion by byjediaz** [ChiLiubio/microeco#104] [Q&A]
Hello, Would you please someone could indicate where  do I get representative sequences for tax4fun 2 analysis?

I don't know from where exactly, in my data could i get these.

Best,

Byron

**Comment by ChiLiubio** [comment]
Hi
Reading representative sequence example is shown here (https://chiliubio.github.io/microeco_tutorial/intro.html#tax4fun2). You can first see the example rep.fna in file2meco package. Actually, no matter which pipeline you use, the representative sequences can be generated along with the ASV/OTU table. So, you can see the software document, such as qiime online tutorial if you are using qiime (https://docs.qiime2.org/2022.2/tutorials/pd-mice/). There is detailed description online about all the important files.

Chi

**Comment by ChiLiubio** [comment]
See https://chiliubio.github.io/microeco_tutorial/file2meco-package.html#qiime2 for directly importing all files including fasta from qiime2.

**Comment by byjediaz** [comment]
@ChiLiubio 

Hello Chi, thanks for your help. 

I used DADA2 pipeline and phyloseq to analyze data. I used a code from [https://github.com/benjjneb/dada2/issues/48 ](url) to obtain rep seqs. I used the phyloseq object to extract microtable objects and added this last file in:
use_dataset <- microtable$new(sample_table = sample_info_16S, otu_table = otu_table_16S, tax_table = taxonomy_table_16S, rep_fasta = rep_fasta_16S)

But, when running: use_dataset$tidy_dataset()

I got the error:

Error in use_dataset$tidy_dataset(main_data = FALSE) : 
  Some feature names are not found in the names of rep_fasta! Please provide a complete fasta file or manually check the names!



Please, i appreciate if you can point me some solution.

Best,

BJ

**Comment by ChiLiubio** [comment]
Hi @byjediaz 
This error message shows that your fasta file is not complete, leading to some ASV can not be found in the fasta file. Generally, the representative fasta is a whole file containing all the ASV/OTU names as it is generated along with the feature table. So it is very strange. Please first check your bioinformatic pipeline.

Best
Chi

**Reply by byjediaz** [reply]
Ok I'll check it out. Thank you.
